Makefile

```diff
@@ -1,15 +1,41 @@
 .ONESHELL:
 
-all: build
+OPENAPI=vmd-api/openapi.yaml
+RUST_SERVER_API=vmd-api/rust-server
+RUST_CLIENT_API=vmd-api/rust-client
 
-generate-sample-certs:
-	make -C test/auth
+CERTS=test/auth/certs
+CLIENT_CERT=$(CERTS)/sample-vmd-client-crt.pem
+CLIENT_KEY=$(CERTS)/sample-vmd-client-key.pem
+CLIENT_CHAIN=$(CERTS)/sample-vmd-client-chain.pem
+SERVER_CERT=$(CERTS)/sample-vmd-server-crt.pem
+SERVER_KEY=$(CERTS)/sample-vmd-server-key.pem
+CA_CERT=$(CERTS)/sample-ca-crt.pem
+
+PORT=8080
+
+# Build
+
+all: generate-server-api generate-client-api build
+
+generate-server-api: $(RUST_SERVER_API)
 
-test-curl-server:
-	curl --cert test/auth/certs/sample-vmd-client-crt.pem	\
-		--key test/auth/certs/sample-vmd-client-key.pem		\
-		--cacert test/auth/certs/sample-ca-crt.pem			\
-		http://localhost:8000/kvm_api_version
+$(RUST_SERVER_API):
```

Comment:
Please add here a comment on why separate generate for server and client are required. That question raises and you had a valid reason for it.

Reply:
https://github.com/tiiuae/vmd/pull/35/commits/13e7d635043fc82cd231bb743ec949659768ab6f